comms/src/macros.rs

```diff
@@ -154,3 +154,14 @@ macro_rules! cfg_next {
         )+
     }
 }
+
+#[macro_export]
+macro_rules! try_spawn_blocking {
```

Comment:
Damn macros. This looks like it shouldn't work since it's returning different types on the match branches. But it does work, doesn't it?

Unless the macro is only used in some very specific places, which makes it kinda not great as a macro, ya know?

Reply:
Agreed, I'll take another pass at flattening the `Result<Result<...>, ....>` where applicable

Reply:
FYI It works because the `v` is a Result type

Reply:
Used a generic utility function called `hoist_nested_result` which is used with `FutureExt::map`